aikau/src/test/resources/alfresco/dnd/NestedReorderTest.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (C) 2005-2015 Alfresco Software Limited.
+ *
+ * This file is part of Alfresco
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @author Dave Draper
+ */
+define(["intern!object",
+        "intern/chai!assert",
+        "require",
+        "alfresco/TestCommon",
+        "intern/dojo/node!leadfoot/keys"], 
+        function (registerSuite, assert, require, TestCommon, keys) {
+
+   var pause = 150;
+   var browser;
+   registerSuite({
+
+      name: "Nested item re-ordering, moving and deleting tests",
+
+      setup: function() {
+         browser = this.remote;
+         return TestCommon.loadTestWebScript(this.remote, "/nested-reorder-dnd", "Nested item re-ordering, moving and deleting tests")
+            .end();
+      },
+
+      beforeEach: function() {
+         browser.end();
+      },
+
+      "Move a deep nested item down a place": function() {
+         return browser.findByCssSelector(".middleTestItemWrapper:nth-child(1) .innerTestItemWrapper:nth-child(1) .down")
+            .click()
+         .end()
+         .clearLog()
+         .findByCssSelector(".confirmationButton > span")
+            .click()
+         .end()
+         .getLastPublish("FORM1_POST")
+            .then(function(payload) {
+               assert.deepPropertyVal(payload, "data.0.config.widgets.0.config.widgets.0.name", "E", "E was not saved as first item");
```

Comment:
Square-bracket notation does work in this assert function, so `data[0].config.widgets[0].config.widgets[0].name` would be equally valid, and more usual for addressing the indices of an array.